lib/core_dom/resource_url_resolver.dart

```diff
@@ -28,8 +28,8 @@ class ResourceUrlResolver {
   static final RegExp urlTemplateSearch = new RegExp('{{.*}}');
   static final RegExp quotes = new RegExp("[\"\']");
 
-  // Ensures that Uri.base is http/https.
-  final _baseUri = Uri.base.origin + ('/');
+  // Reconstruct the Uri without the http or https restriction due to Uri.base.origin 
+  final _baseUri = "${Uri.base.scheme}://${Uri.base.authority}" + ('/');
```

Comment:
Check if `Uri.base.authority` is empty and throw and error if it is.  To do that, you might make this assigned the result of a calling a static function.  (It will be initialized lazily so it won't throw an error when relative url support is disabled.)  Otherwise, looks good.